src/nnet3/nnet-diagnostics.h

```diff
@@ -108,7 +118,7 @@ class NnetComputeProb {
   void Compute(const NnetExample &eg);
 
   // Prints out the final stats, and return true if there was a nonzero count.
-  bool PrintTotalStats() const;
```

Comment:
You don't say what the semantics of 'tot_weight' are and what it represents.  As the code is written, it appears to *add* to 'total_weight'.  It's not clear from the interface that this is the intended behavior.

Also you seems to have changed the semantics of PrintTotalStats' return value.  As it was, it returned true if the count for one of the outputs of the nnet was nonzero.  This does the right thing in, for instance, the multilingual training setup.  You seem to have changed it  to return true if they are *all* nonzero, and this won't always be the right thing.  Certainly it's inconsistent with the documentation.